### PR TITLE
If the original table is MyISAM and the default engine is Innodb, and…

### PR DESCRIPTION
**Pull request by rj03hou**
Related issue: https://github.com/github/gh-ost/issues/320

The following is all done.
- [x] contributed code is using same conventions as original code
- [x] code is formatted via `gofmt` (please avoid `goimports`)
- [x] code is built via `./build.sh`
- [x] code is tested via `./test.sh`

### Description

just make sure the the magic cut-over table's engine same with the original table.
detail description is in https://github.com/github/gh-ost/issues/320.

> In case this PR introduced Go code changes:

… the gtid mode is on, there will be error when execute 'LOCK TABLES tbl WRITE, tbl_magic WRITE'. If make the magic cut-over table's engine same with the original table, there will not have this problem.

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Looking good. Thank you

File: go/logic/applier.go [hunk 1]
```diff
@@ -703,8 +703,9 @@ func (this *Applier) CreateAtomicCutOverSentryTable() error {
 
 	query := fmt.Sprintf(`create /* gh-ost */ table %s.%s (
 			id int auto_increment primary key
-		) comment='%s'
+		) engine=%s comment='%s'
 		`,
+		this.migrationContext.TableEngine,
```
Comment (shlomi-noach) [inline go/logic/applier.go]: `TableEngine` correlates to the 3rd placeholder in the template string, not the 1st

**Comment by rj03hou** [comment]
I change the order, and make the test. 
sorry about this, I make change and test in one environment, and push the code another environment.

I unify the environment, next time, will be ok.

**Comment by shlomi-noach** [comment]
Resubmitted as https://github.com/github/gh-ost/pull/325

**Pull request merged**